packages/currency/src/currencyManager.ts

```diff
@@ -71,17 +71,21 @@ export class CurrencyManager<TMeta = unknown> implements ICurrencyManager<TMeta>
       return this.fromAddress(currencyIdentifier, network);
     }
 
+    if (network && currencyIdentifier.indexOf(network) === -1) {
+      currencyIdentifier = `${currencyIdentifier}-${network}`;
+    }
```

Comment:
In making the improvement below, we need to handle the case where we provide the network as separate parameter but the network is not within the symbol. 
There is an issue when we have both Fiat and ERC20 token with same symbol, like MNT.
MNT - fiat currency
MNT-mainnet - Ethereum token.
The simplest fix here was to add the network to the currencyIdentifier if it's not already there.